presto-client/src/main/java/com/facebook/presto/client/GCSOAuthHandler.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.client;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import okhttp3.Authenticator;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.Route;
+
+import javax.annotation.Nullable;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collections;
+
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_EXTRA_CREDENTIAL;
+import static java.util.Objects.requireNonNull;
+
+public class GCSOAuthHandler
+        implements Interceptor, Authenticator
+{
+
+    public static final String GCS_CREDENTIALS_PATH_KEY = "hive.gcs.credentials.path";
```

Comment:
shall we keep hive here? kind of misleading. If the session property is already defined, could we import the string from other class?

Reply:
The "hive.gcs.credentials.path" is the key I newly added.   The other 'hive.gcs.oauth' is an existing one, which can be also found in presto-hive. 

presto-hive/src/main/java/com/facebook/presto/hive/gcs/GcsConfigurationProvider.java
28:    private static final String GCS_OAUTH_KEY = "hive.gcs.oauth";

If we imported the constant "hive.gcs.oauth" from presto-hive,  we might introduce a new dependency from presto-client to presto-hive.  Shall we do that?

I'm just thinking the name of these two keys need to be consistent.

Reply:
get it. no need to have dependency from presto-hive
keep `hive.gcs.credentials.path` is fine